src/ts/components/subtitleoverlay.ts

```diff
@@ -416,3 +422,97 @@ class ActiveSubtitleManager {
     this.activeSubtitleCueCount = 0;
   }
 }
+
+export class SubtitleRegionContainerManager {
+  private subtitleRegionContainers: { [regionName: string]: SubtitleRegionContainer } = {};
+  private labelsInRegionCount: { [regionName: string]: number } = {};
+
+  /**
+   * @param subtitleOverlay Reference to the subtitle overlay for adding and removing the containers.
+   */
+  constructor(private subtitleOverlay: SubtitleOverlay) {
+    this.subtitleOverlay = subtitleOverlay;
+  }
+
+  /**
+   * Removes a subtitle label from a container.
+   */
+  removeLabel(labelToRemove: SubtitleLabel, event?: SubtitleCueEvent): void {
+    const region = event && event.region || 'default';
+    for (const regionName in this.subtitleRegionContainers) {
+      this.subtitleRegionContainers[regionName].removeLabel(labelToRemove);
+    }
+    this.labelsInRegionCount[region]--;
+
+    // Remove container if no more labels are displayed
+    if (this.labelsInRegionCount[region] === 0) {
+      this.subtitleOverlay.removeComponent(this.subtitleRegionContainers[region]);
+      delete this.subtitleRegionContainers[region];
+    }
+  }
+
+  /**
+   * Removes all subtitle containers.
+   */
+  clear(): void {
+    for (const regionName in this.subtitleRegionContainers) {
+      this.subtitleOverlay.removeComponent(this.subtitleRegionContainers[regionName]);
+    }
+
+    this.subtitleRegionContainers = {};
+    this.labelsInRegionCount = {};
+  }
+
+  /**
+   * Creates and wraps a subtitle label into a container div based on the subtitle region.
+   * If the subtitle has positioning information it is added to the container.
+   * @param label The subtitle label to wrap
+   * @param event Subtitle cue event that contains the positoning information
+   */
+  addLabel(label: SubtitleLabel, event?: SubtitleCueEvent): void {
+    const regionName = event && event.region || 'default';
```

Comment:
As noted above, it shouldn't be necessary to provide the `event` here and also in `removeLabel`.